OrbitLinuxTracing/TracerThread.h

```diff
@@ -28,16 +27,15 @@
 #include "Utils.h"
 #include "absl/container/flat_hash_map.h"
 #include "absl/container/flat_hash_set.h"
+#include "capture.pb.h"
 
 namespace LinuxTracing {
 
 class TracerThread {
  public:
-  TracerThread(pid_t pid, uint64_t sampling_period_ns,
-               std::vector<Function> instrumented_functions)
-      : pid_(pid),
-        sampling_period_ns_(sampling_period_ns),
-        instrumented_functions_(std::move(instrumented_functions)) {}
+  static constexpr double DEFAULT_SAMPLING_FREQUENCY = 1000.0;
```

Comment:
kDefaultSampling....

Reply:
Also this seems like a fallback_frequency... something we use if user specified invalid one? Add a comment maybe?
Also once we have it provided as part of the request we should probably return a error instead of falling back to default frequency.

Reply:
So, if we write error to the log and fallback on some value - nobody will notice the error. I think we should be explicit here and fail with clear message of what went wrong instead of doing something user does not expect.

Reply:
Bubbling up an error from here all the way to the client is not an option with the current code. We might want to change that but that's out of the scope of this already very big change. For this specific case, the validity of the sampling rate should be checked all the way back in the client once it's possible to specify it, this is an extra check internal to the module.

Reply:
Ok, if it is checked on the client - that means we can probably just abort if this is incorrect. CHECK or FAIL here and remove fallback?

Reply:
Sounds fair, used `FAIL_IF`.